From: Rene Kita <mail@rkta.de>
Subject: Replace deprecated MD5 function

OpenSSL deprecated MD5 with Release 3.0. Provide a wrapper function around
the new way to do it.

As the old code did no error handling this patch does not do it either.

---
 file.c |   16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

--- a/file.c
+++ b/file.c
@@ -1186,7 +1186,7 @@ AuthBasicCred(struct http_auth *ha, Str
 }
 
 #ifdef USE_DIGEST_AUTH
-#include <openssl/md5.h>
+#include <openssl/evp.h>
 
 /* RFC2617: 3.2.2 The Authorization Request Header
  * 
@@ -1213,6 +1213,20 @@ AuthBasicCred(struct http_auth *ha, Str
  *                     "c" | "d" | "e" | "f"
  */
 
+#define MD5_DIGEST_LENGTH 16
+
+static void
+MD5(const unsigned char *d, unsigned long n, unsigned char *md)
+{
+    EVP_MD_CTX *ctx = EVP_MD_CTX_new();
+
+    EVP_DigestInit_ex(ctx, EVP_md5(), NULL);
+    EVP_DigestUpdate(ctx, d, n);
+    EVP_DigestFinal_ex(ctx, md, NULL);
+    EVP_MD_CTX_free(ctx);
+}
+
+
 static Str
 digest_hex(unsigned char *p)
 {
